sdk/communication/azure-communication-chat/src/main/java/com/azure/communication/chat/implementation/CommunicationBearerTokenCredential.java

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.communication.chat.implementation;
+
+import java.util.concurrent.ExecutionException;
+
+import com.azure.communication.common.CommunicationTokenCredential;
+import com.azure.core.credential.AccessToken;
+import com.azure.core.credential.TokenCredential;
+import com.azure.core.credential.TokenRequestContext;
+
+import reactor.core.publisher.Mono;
+
+/**
+ * This class serves as a CommunicationTokenCredential wrapper that 
+ * allows using BearerAuthenticationPolicy in different clients
+ */
+public class CommunicationBearerTokenCredential implements TokenCredential {
+    private final CommunicationTokenCredential credential;
+
+    /**
+     * Creates a CommunicationTokenCredential
+     *
+     * @param communicationTokenCredential The {@link CommunicationTokenCredential} to use 
+     * in the BearerAuthenticationPolicy.
+     */
+    public CommunicationBearerTokenCredential(CommunicationTokenCredential communicationTokenCredential) {
+        credential = communicationTokenCredential;
+    }
+
+    @Override
+    public Mono<AccessToken> getToken(TokenRequestContext request) {
+        try {
+            return Mono.just(credential.getToken().get());
```

Comment:
Is this because `CommunicationTokenCredential.getToken` returns a `Future` instead of a `Promise`?

Reply:
Yes, I am currently prepping another PR to address that issue